### PR TITLE
breaking: remove platform-centered workflow

### PR DESCRIPTION
**Pull request by erisu**
### Motivation and Context

Remove the broken platform-centered workflow from Cordova Electron.

Only create and version worked but the essentials such as build and run was not supported.

IMO, it is easier to support and focus on a single workflow for creating and building a Cordova Electron application. The CLI workflow is documented and generally preferred.

The platform-centered workflow is only briefly mentioned and has some documentation for other platforms but non of them work for Electron.

As Electron is a newer platform and does not support this process, its safer to remove the broken, incomplete, and partial implemented workflow instead of take effort to support something that isn't fully documented.

### Description

* Delete all binaries, except for `version` which is still used by `cordova-lib`
* Enforce config parser argument to exist when `createPlatform` is called. This argument is always supplied by CLI workflow.

### Testing

- `npm t`
- `cordova platform add`
- `cordova platform` -> checks the versions
- `cordova build electron`
- `cordova run electron --nobuild`

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [x] I added automated test coverage as appropriate for this change

### CODE REVIEW

**Comment by erisu** [comment]
Last call for reviews. This PR will be merged in 24 hours.

**Comment by erisu** [comment]
Platform-Centered workflow has never worked with Electron and will never be implemented. IT will be confusing to keep broken code laying around that is never used.

**Comment by NiklasMerz** [comment]
I agree that removing the platform centered workflow should be discussed and done carefully for all other platforms.

For cordova-electron the situation looks different to me. cordova-electron is a newer platform which is quite different. This code never really worked and therefore nobody should use it now.

If someone picks up and fixes this we could it later if it makes sense.

**Comment by brodybits** [comment]
:+1:, now adding reference to the general discussion in <https://github.com/apache/cordova/issues/198>.

File: tests/spec/fixtures/test-config-empty.xml [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
 -->
-
-<widget>
+<widget xmlns="http://www.w3.org/ns/widgets">
     <name />
     <description />
     <author />
     <content />
-</widget>
+</widget>
```
Comment (brodybits) [inline tests/spec/fixtures/test-config-empty.xml]: Looks like an editor removed the final newline here

File: tests/spec/fixtures/test-config-empty.xml [hunk 2]
```diff
@@ -1,26 +1,25 @@
-<?xml version='1.0' encoding='utf-8'?>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (brodybits) [inline tests/spec/fixtures/test-config-empty.xml]: I do not (yet) see the relevance of this update in terms of testing. Any other thoughts here?

File: tests/spec/fixtures/test-config-empty.xml [hunk 3]
```diff
@@ -1,26 +1,25 @@
...
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
 -->
-
-<widget>
+<widget xmlns="http://www.w3.org/ns/widgets">
```
Comment (brodybits) [inline tests/spec/fixtures/test-config-empty.xml]: Ditto: I do not really see the relevance of this update for the purpose of testing. Any other thoughts here:

**Comment by erisu** [comment]
@brodybits The tests were added to keep full coverage after removing the `create` bin.

The `create` bin created a config parser object that covered one use case. Removing the bin removed that test. Because of this drop in coverage, I believe it is expected that it should not be lost.

The test itself maybe could be pulled out in this case but we should never be strict to tell a user to make multiple PRs just to be able to merge in their changes. It seems wrong that if their change causes coverage to drop, they should create a second PR to PRE-fix the drop of coverage before accepting changes.

**Comment by erisu** [comment]
I went ahead and updated the code to follow inline with CLI workflow which now forced the test to be changed as well.

**Comment by brodybits** [comment]
Thanks @erisu. I was actually working on my response when you pushed the other update. I may need a little more time to review and understand the API changes, unless you want me to leave the reviews to someone else.

In short, I definitely like the idea of keeping related changes in the same PR, as long as they are in separate commits with clean messages, ready to be squashed together if appropriate. (I would definitely favor doing a squash merge in this PR.) This makes it easier for me and others to understand the story flow of what happened, both in the git history and in case people want to look more closely through the PR in the future.

For this PR, I would favor adding a quick phrase to the title about the enforced API change.

I would also favor keeping most of the originally proposed syntax cleanup changes in one or more "cleanup" commits. The one thing I didn't like was when the editor removed the newline at the end of a file.

This is how I have been working with some other projects, which seem to *always* do squash merge.

**Comment by erisu** [comment]
I already updated the entire PR title and description. It also includes

> Enforce config parser argument to exist when `createPlatform` is called. This argument is always supplied by CLI workflow.

**Pull request merged**